server/konnectors/linkedin.coffee

```diff
@@ -0,0 +1,328 @@
+        for contact in contacts
+            account = contact.getAccount ACCOUNT_TYPE, entries.accountName
+            if account?
+                entries.ofAccountByIds[account.id] = contact
+                entries.ofAccount.push contact
+        next()
+
+
+retrieveContacts = (requiredFields, entries, data, next) ->
+
+    processRetrievingContactData = (contact, next) ->
+        contacts = "https://www.linkedin.com/contacts/api/contacts/#{contact.id}/?fields=name,first_name,last_name,emails_extended,phone_numbers,sites,addresses,company,title,location,profiles,twitter,display_sources&_=1444864592009"
+
+        opts =
+            url: contacts
+            jar: true
+            json: true
+            headers:
+                'User-Agent': agent
+
+        request.get opts, (err, res, body) ->
+            return next err if err
+
+            if body.status? and body.status == 'error'
+                next new Error(body.status_details)
+            datapoints = []
+
+            data = body.contact_data
+            datapoints = datapoints.concat getPhoneNumber(data)
+            datapoints = datapoints.concat getEmails(data)
+            datapoints = datapoints.concat getUrls(data)
+            datapoints = datapoints.concat getAddresses(data)
+
+
+            finalContact = new Contact
+                n: "#{data.last_name};#{data.first_name}"
+                fn: data.name
+                title: contact.title || undefined
+                org: data.company?.name || undefined
+                title: data.title || undefined
+                tags: ['linkedin']
+                datapoints: datapoints || undefined
+
+            finalContact.imageUrl = contact.secure_profile_image_url || undefined
+            ContactHelper.setAccount finalContact,
+                type: ACCOUNT_TYPE
+                name: entries.accountName
+                id: data.id
+
+            next null, finalContact
+
+
+    queue = async.queue processRetrievingContactData, 10
+
+    queue.drain = ->
+        log.info 'All data retrieved'
+        next()
+
+    Tag.getOrCreate { name: 'linkedin', color: '#1B86BC'}, (err, tag) ->
+        if err
+            entries.tag = null
+        else
+            entries.tag = tag
+        entries.listContacts.forEach (contact) ->
+            queue.push contact, (err, finalContact) ->
+                log.error err if err
+                saveContact finalContact, entries
+
+
+
+
+
+getPhoneNumber = (data) ->
+    listPhones = []
+
+    data.phone_numbers.forEach (number) ->
+        listPhones.push
+            name: 'tel'
+            type: number.type.toLowerCase()
+            value: number.number.replace(/ /g, '')
+    listPhones
+
+
+getEmails = (data) ->
+    listEmails = []
+
+    data.emails_extended.forEach (email) ->
+        listEmails.push
+            name: 'email'
+            value: email.email
+            type: 'internet'
+            pref: email.primary == true ? true : undefined
```

Comment:
I'm surprised that coffeescript accepts this line. In fact, it does something really funny, but not what is wanted.